typescript/infra/scripts/warp-routes/monitor-warp-routes-balances.ts

```diff
@@ -0,0 +1,131 @@
+import { Connection } from '@solana/web3.js';
+import { ethers } from 'ethers';
+import { Gauge, Registry } from 'prom-client';
+
+import {
+  ERC20__factory,
+  SealevelHypCollateralAdapter,
+} from '@hyperlane-xyz/hyperlane-token';
+import { ChainMap, MultiProvider } from '@hyperlane-xyz/sdk';
+import { debug, objMap, promiseObjAll } from '@hyperlane-xyz/utils';
+
+import { startMetricsServer } from '../../src/utils/metrics';
+
+import { WarpTokenConfig, tokenList } from './token_config';
+
+const metricsRegister = new Registry();
+const warpRouteTokenBalance = new Gauge({
+  name: 'hyperlane_warp_route_token_balance',
+  help: 'HypERC20 token balance of a Warp Route',
+  registers: [metricsRegister],
+  labelNames: [
+    'chain_name',
+    'token_address',
+    'token_name',
+    'wallet_address',
+    'token_type',
+  ],
+});
+
+async function main(): Promise<boolean> {
+  startMetricsServer(metricsRegister);
+
+  const checkFreqeuncy = 1000; // 1 second
+
+  const multiProvider = new MultiProvider();
+
+  setInterval(async () => {
+    console.log('Checking balances');
+    const balances = await checkBalance(tokenList, multiProvider);
```

Comment:
can we wrap this call with a try / catch and log an error if one occurs?

The nautilus RPC URL is proving to be really flaky and is causing the deployed warp route monitoring to crash really frequently, e.g.:

```
$ kubectl logs hyperlane-warp-route-zebec-0
...
...

Checking balances
/hyperlane-monorepo/node_modules/@ethersproject/web/node_modules/@ethersproject/logger/src.ts/index.ts:269
        const error: any = new Error(message);
                           ^
Error: bad response (status=504, headers={"content-length":"24","content-type":"text/plain","via":"1.1 google","date":"Tue, 29 Aug 2023 10:07:02 GMT","alt-svc":"h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000","connection":"close"}, body="upstream request timeout", requestBody="{\"method\":\"eth_getBalance\",\"params\":[\"0x4501bbe6e731a4bc5c60c03a77435b2f6d5e9fe7\",\"latest\"],\"id\":45,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="https://api.nautilus.nautchain.xyz", code=SERVER_ERROR, version=web/5.7.0)
    at Logger.makeError (/hyperlane-monorepo/node_modules/@ethersproject/web/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
    at Logger.throwError (/hyperlane-monorepo/node_modules/@ethersproject/web/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
    at /hyperlane-monorepo/node_modules/@ethersproject/web/src.ts/index.ts:302:24
    at step (/hyperlane-monorepo/node_modules/@ethersproject/web/lib/index.js:33:23)
    at Object.next (/hyperlane-monorepo/node_modules/@ethersproject/web/lib/index.js:14:53)
    at fulfilled (/hyperlane-monorepo/node_modules/@ethersproject/web/lib/index.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  reason: 'bad response',
  code: 'SERVER_ERROR',
  status: 504,
  headers: {
    'content-length': '24',
    'content-type': 'text/plain',
    via: '1.1 google',
    date: 'Tue, 29 Aug 2023 10:07:02 GMT',
    'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000',
    connection: 'close'
  },
  body: 'upstream request timeout',
  requestBody: '{"method":"eth_getBalance","params":["0x4501bbe6e731a4bc5c60c03a77435b2f6d5e9fe7","latest"],"id":45,"jsonrpc":"2.0"}',
  requestMethod: 'POST',
  url: 'https://api.nautilus.nautchain.xyz'
}
```